nexus-repository-conan/src/main/java/org/sonatype/repository/conan/internal/proxy/v1/ConanProxyFacet.java

```diff
@@ -286,19 +287,43 @@ protected void indicateVerified(final Context context, final Content content, fi
   protected String getUrl(@Nonnull final Context context) {
     AssetKind assetKind = context.getAttributes().require(AssetKind.class);
 
-    if (DOWNLOAD_ASSET_KINDS.contains(assetKind)) {
+    if (METADATA_ASSET_KINDS.contains(assetKind)) {
       return context.getRequest().getPath().substring(1);
     }
 
+    TokenMatcher.State state = context.getAttributes().require(TokenMatcher.State.class);
+    ConanCoords coords = convertFromState(state);
+
+    if (assetKind == CONAN_MANIFEST) {
+      return getUrlForConanManifest(coords);
+    }
+
     log.info("AssetKind {} to be fetched is {}", assetKind, context.getRequest().getPath());
 
     // TODO: There are two different URLs for DOWNLOAD_URL, this seems to only look in one of them, that seems problematic
-    TokenMatcher.State state = context.getAttributes().require(TokenMatcher.State.class);
-    ConanCoords coords = convertFromState(state);
     String download_urls = ConanProxyHelper.getProxyAssetPath(coords, DOWNLOAD_URL);
     return getUrlFromDownloadAsset(download_urls, assetKind.getFilename());
   }
 
+  @TransactionalTouchBlob
+  @Nullable
+  protected String getUrlForConanManifest(final ConanCoords coords) {
+    String downloadUrlsAssetPath = ConanProxyHelper.getProxyAssetPath(coords, DOWNLOAD_URL);
+    StorageTx tx = UnitOfWork.currentTx();
+
+    Asset downloadUrlAsset = findAsset(tx, tx.findBucket(getRepository()), downloadUrlsAssetPath);
+    if (downloadUrlAsset == null) {
+      // TODO Looks like it was search request. So let's look up conanmanifest url from digest
```

Comment:
Add ticket number to TODO

Reply:
fixed https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/98/commits/3c7d54608b90e0730d43d62d549032b70bb4d380